hw_diag/templates/diagnostics_page.html

```diff
@@ -92,38 +66,46 @@ <h3 class="text-center mb-4">Diagnostics Breakdown</h3>
                 </tr>
                 <tr>
                   <td>Region Plan</td>
-                  {% if diagnostics.RE == 'UN123' %}
-                    <td class="text-right">Awaiting Location Assertion</td>
+                  {% if diagnostics.gatewayrs.RE %}
+                    <td class="text-right">{{ diagnostics.gatewayrs.RE }}</td>
                   {% else %}
-                    <td class="text-right">{{ diagnostics.RE }}</td>
+                    <td class="text-right">Unknown Location</td>
                   {% endif %}
                 </tr>
                 <tr>
                   <td>Variant</td>
                   <td class="text-right">{{ diagnostics.FRIENDLY }}</td>
                 </tr>
                 <tr>
-                  <td>Miner Connected To Blockchain</td>
-                  {% if diagnostics.MC %}
-                    <td class="text-success text-right">True</td>
+                  <td>Connected Validator</td>
+                  {% if diagnostics.gatewayrs.validator_uri %}
```

Comment:
not checking for diagnostics.gatewayrs for validity. As far as I can see DiagnosticReport make sure that they key is there in case of failure also. 

Let me know if I am mistaken.

Reply:
You probably want to use [check_errors](https://github.com/NebraLtd/hm-pyhelper/blob/b48f500b407d3c2ce8651844bb8a939746847691/hm_pyhelper/diagnostics/diagnostics_report.py#L49). If a Diagnostic calls record_result, then the key maps to a valid value. If it calls record_failure, then the key maps to an error value.

Reply:
Here I couldn't use check errors as render_template is passed json read from the file. It is not an instance of Diagnostic. But Will try and see if something else can be done. For now, It will show actual error that happened instead of "Disconnected". Disconnected will be shown only when the key itself is missing from json.